spec/infinispan_cluster_spec.js

```diff
@@ -30,11 +33,467 @@ describe('Infinispan cluster client', function() {
+  });
+
+  xit('can iterate over entries getting their expirable metadata', function(done) {
+    var pairs = [{key: 'it-exp-1', value: 'v1'}, {key: 'it-exp-2', value: 'v2'}];
+    var expected = _.map(pairs, function(pair) {
+      return f.merge(pair, {done: false, lifespan: 86400, maxIdle : 3600});
+    });
+    client
+        .then(t.assert(t.putAll(pairs, {lifespan: '1d', maxIdle: '1h'}), t.toBeUndefined))
+        .then(function(client) {
+          return client2
+              .then(t.parIterator(1, expected, {metadata: true})) // Iterate all data, 1 element at time, parallel
+              .then(function() {
+                return client;
+              });
+        })
+        .then(function(client) {
+          return client3
+              .then(t.seqIterator(3, expected, {metadata: true})) // Iterate all data, 1 element at time, parallel
+              .then(function() {
+                return client;
+              });
+        })
+        .catch(t.failed(done))
+        .finally(done);
+  });
+
+  it('can execute a script remotely to store and retrieve data in local mode using different clients', function(done) {
+    Promise.all([client, readFile('spec/utils/typed-put-get.js')])
+        .then(function(vals) {
+          var c = vals[0];
+          return c.addScript('typed-put-get.js', vals[1].toString())
+              .then(function() { return c; } );
+        })
+        .then(t.assert(t.exec('typed-put-get.js', {k: 'typed-key', v: 'typed-value'}),
+            t.toBe('typed-value')))
+        .then(function(client) {
+          return client3
+              .then(t.assert(t.exec('typed-put-get.js', {k: 'typed-key', v: 'typed-value3'}), t.toBe('typed-value3')))
+              .then(function() {
+                return client;
+              })
+        })
+        .then(function(client) {
+          return client2
+              .then(t.assert(t.exec('typed-put-get.js', {k: 'typed-key', v: 'typed-value2'}), t.toBe('typed-value2')))
+              .then(function() {
+                return client;
+              })
+        })
+        .catch(t.failed(done)).finally(done);
+  });
+
+  xit('can execute a script remotely to store and retrieve data in distributed mode', function(done) {
+    Promise.all([client, readFile('spec/utils/typed-put-get-dist.js')])
+        .then(function(vals) {
+          var c = vals[0];
+          return c.addScript('typed-put-get-dist.js', vals[1].toString())
+              .then(function() { return c; } );
+        })
+        .then(t.assert(t.exec('typed-put-get-dist.js', {k: 'typed-key', v: 'typed-value'}),
+            t.toContain(['typed-value', 'typed-value', 'typed-value'])))
+        .catch(t.failed(done)).finally(done);
+  });
+
   // Since Jasmine 1.3 does not have afterAll callback, this disconnect test must be last
   it('disconnects client', function(done) { client
       .then(t.disconnect())
       .catch(t.failed(done))
       .finally(done);
+
+    client2
```

Comment:
If the first client fails to disconnect this won't be invoked and neither client3's disconnect method. Instead, this should be written in such way that you disconnect first client, then disconnect second client... and a single catch + finally section.